Makefile

```diff
@@ -39,8 +39,18 @@ migration:
 	# building migration
 	go build -o bin/machine-api-migration cmd/machine-api-migration/main.go
 
-unit:
-	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) -p path --bin-dir $(PROJECT_DIR)/bin --index https://raw.githubusercontent.com/openshift/api/master/envtest-releases.yaml)" ./hack/test.sh "./pkg/... ./manifests-gen/..." 5m
+.PHONY: localtestenv
+localtestenv: .localtestenv
+
+KUBEBUILDER_ASSETS ?= $(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) -p path --bin-dir $(PROJECT_DIR)/bin --index https://raw.githubusercontent.com/openshift/api/master/envtest-releases.yaml)
+.localtestenv: Makefile
+	echo "KUBEBUILDER_ASSETS=$(KUBEBUILDER_ASSETS)" > $@
+
+TEST_DIRS ?= ./pkg/... ./manifests-gen/...
+unit: .localtestenv
+	# Launch "./hack/test.sh ..." with additional environment variables
+	# loaded from .localtestenv
+	xargs -I '{}' -x -- env '{}' ./hack/test.sh "$(TEST_DIRS)" 5m < .localtestenv
```

Comment:
We need to make sure this is compatible with standard macOS xargs (BSD) vs GNU.

Reply:
@damdo Can you run it for me? `make unit` should do the trick.

Reply:
I run nixpks one, so I use the gnu version by default. Better to ask @JoelSpeed or @theobarberbany

Reply:
`xargs -I '{}' -x env '{}' ./hack/test.sh "./pkg/... ./manifests-gen/..." 5m < .localtestenv` ?

Reply:
What is breaking exactly? looks like both `-x` and `-I` are there?

Reply:
Looks like it's the `--` separator

Reply:
Unfortunately `-n1` wouldn't be correct here, as it would transform this into a separate test invocation for each local env var.

Reply:
I preferred the 'just use gxargs on Mac' solution: it sounds much more maintainable. Shall we add something to the Makefile which detects `gxargs` and uses it in preference to `xargs` if it's present?

Reply:
Sounds the next best solution then yes